modules/core/src/main/java/org/apache/synapse/unittest/ConfigurationDeployer.java

```diff
@@ -0,0 +1,199 @@
+        configurationContext.setAxisConfiguration(axisConfiguration);
+
+        //initialize sequence deployer using created configuration context
+        sequenceDeployer.init(configurationContext);
+
+        //deploy synapse artifact
+        String deployedArtifact = sequenceDeployer.deploySynapseArtifact(inputElement, fileName, null);
+
+        return new Pair<>(synapseConfiguration, deployedArtifact);
+    }
+
+    /**
+     * Method of deploying proxy artifact in synapse.
+     *
+     * @param inputElement synapse configuration artifact as OMElement type
+     * @param fileName     name of the file
+     * @return response of the artifact deployment and the synapse configuration as a Pair<>
+     */
+    Pair<SynapseConfiguration, String> deployProxyArtifact(OMElement inputElement, String fileName)
+            throws AxisFault {
+
+        //create new proxy service deployer object
+        ProxyServiceDeployer proxyServiceDeployer = new ProxyServiceDeployer();
+
+        //create a synapse configuration and set all axis2 configuration to it
+        SynapseConfiguration synapseConfiguration = UnitTestingExecutor.getExecuteInstance().getSynapseConfiguration();
+        AxisConfiguration axisConfiguration = synapseConfiguration.getAxisConfiguration();
+        ConfigurationContext configurationContext = new ConfigurationContext(axisConfiguration);
+        SynapseEnvironment synapseEnvironment = new Axis2SynapseEnvironment(configurationContext, synapseConfiguration);
+        axisConfiguration.addParameter(new Parameter(SynapseConstants.SYNAPSE_ENV, synapseEnvironment));
+        axisConfiguration.addParameter(new Parameter(SynapseConstants.SYNAPSE_CONFIG, synapseConfiguration));
+        configurationContext.setAxisConfiguration(axisConfiguration);
+
+        //initialize proxy service deployer using created configuration context
+        proxyServiceDeployer.init(configurationContext);
+
+        //deploy synapse artifact
+        String deployedArtifact = proxyServiceDeployer.deploySynapseArtifact(inputElement, fileName, null);
+
+        return new Pair<>(synapseConfiguration, deployedArtifact);
+    }
+
+    /**
+     * Method of deploying API artifact in synapse.
+     *
+     * @param inputElement synapse configuration artifact as OMElement type
+     * @param fileName     name of the file
+     * @return response of the artifact deployment and the synapse configuration as a Pair<>
+     */
+    Pair<SynapseConfiguration, String> deployApiArtifact(OMElement inputElement, String fileName)
+            throws AxisFault {
+
+        //create new API deployer object
+        APIDeployer apiResourceDeployer = new APIDeployer();
+
+        //create a synapse configuration and set all axis2 configuration to it
+        SynapseConfiguration synapseConfiguration = UnitTestingExecutor.getExecuteInstance().getSynapseConfiguration();
```

Comment:
Shall we extract the duplicate code segments to a separate method?

Reply:
Fixed in https://github.com/wso2/wso2-synapse/pull/1452